### ISSUE TITLE
trial verbose color reporter should colorize final "PASSED" and "FAILED" message

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2856]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2856|
|Type|enhancement|
|Created|2007-10-24 15:00:28Z|

It'd be much easier to read the summary output if "PASSED" were green and "FAILED" were red, or maybe if the entire lines were colored according to that status information.

<details><summary>Searchable metadata</summary>

```
trac-id__2856 2856
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1193238028000000 1193238028000000
changetime__1193317393000000 1193317393000000
version__None None
owner__ 
cc__therve cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [21407]) Color the ouput and add tests for it.

Refs [#2856](https://github.com/twisted/twisted/issues/2856)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

That's a nice idea, implemented and ready to review in trial-colored-summary-2856. I choose to color only the main message, not the whole line.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @therve|
|-|-|

It's too bad Reporter still has that stream attribute. :(

printSummary doesn't seem to be part of any interface.  I guess this is the reporter object.  The runner calls ```result.printSummary()```, anyway, and I guess ```result``` is supposed to provide ```IReporter```.  But that's an existing problem, not a new one.

Doing ```if not wasSuccessful```/```FAILED``` instead of ```if wasSuccessful```/```PASSED``` is slightly round-about, but I see that's just how the parent implementation being overridden works.  Plus it's not a very big deal in the first place...

I'm just nit picking basically, the branch looks great.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [21410]) Merge trial-colored-summary-2856

Author: therve
Reviewer: exarkun
Fixes [#2856](https://github.com/twisted/twisted/issues/2856)

Make the trial colored verbose reporter colorize final PASSED and FAILED
messages.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
